lib/engine/game/g_1849/game.rb

```diff
@@ -436,6 +466,12 @@ def close_corporation(corporation, quiet: false)
           end
           corporation.next_to_par = true if @corporations[index - 1].floated?
           update_garibaldi
+          # code below is for variant rules
+          @log << 'E-token returned to bank' if corporation.e_token
+          corporation.e_token = false if electric_dreams?
+
+          repaid_bond = corporation.loans.pop if bonds? && corporation.loans.any?
```

Comment:
Here also, don't need to check bonds. If the corporation has loans, that tells us bonds are enabled. `corporation.loans.clear()`.